src/tasks/index.ts

```diff
@@ -112,336 +104,147 @@ export default new class TaskQueue {
     return queue;
   }
 
-  /**
-   * Adds task to contractDocumentStatusQueue
-   *
-   * @param {int} contractDocumentId id
-   */
-  enqueueContractDocumentStatusTask(contractDocumentId: number) {
-    this.contractDocumentStatusQueue.push({id: contractDocumentId, contractDocumentId: contractDocumentId});
-  }
-
-  /**
-   * Adds task to contractDocumentStatusBatchQueue
-   */
-  enqueueContractDocumentStatusBatchQueue() {
-    this.contractDocumentStatusBatchQueue.push({id: 1});
-  }
-
   /**
    * Adds task to sapContractDeliveredQuantityUpdateQueue
    */
-  private enqueueContractDeliveredQuantityUpdateQueue() {
-    this.sapContractDeliveredQuantityUpdateQueue.push({id: 1});
+  private enqueueContractDeliveredQuantityUpdateQueue = () => {
+    this.sapContractDeliveredQuantityUpdateQueue.push({ id: 1 });
   }
 
   /**
-   * Enqueues SAP contact update task
+   * Enqueues SAP business partners update task
    *
-   * @param {Object} businessPartner SAP business partner object
+   * @param operationReportId operation report ID when task is triggered manually
    */
-  async enqueueSapContactUpdate(operationReportId: number, businessPartner: SapBusinessPartner) {
-    const operationReportItem = await models.createOperationReportItem(operationReportId, null, false, false);
-
+  public enqueueSapContactUpdate = async (operationReportId?: number) => {
     this.sapContactUpdateQueue.push({
-      id: businessPartner.CardCode,
-      businessPartner: businessPartner,
-      operationReportItemId: operationReportItem.id
+      id: 1,
```

Comment:
It is 1, since the queue library will automatically cancel the previous task, when a task with the same ID is pushed to the queue. I use this to cancel the possible already existing task so that there is not ever two of these tasks at the same time in the queue. This is required because the task pushes a new task to the queue in the end.